src/dds/rsdds-device-factory.cpp

```diff
@@ -163,6 +172,43 @@ std::vector< std::shared_ptr< device_info > > rsdds_device_factory::query_device
     {
         unsigned const mask = context::combine_device_masks( requested_mask, get_context()->get_device_mask() );
 
+        auto participant = _watcher_singleton->get_device_watcher()->get_participant();
+        domain_context * p_domain = nullptr;
+        {
+            std::lock_guard< std::mutex > lock( domain_context_by_id_mutex );
+            auto it = domain_context_by_id.find( participant->domain_id() );
+            if( it != domain_context_by_id.end() )
+                p_domain = &it->second;
+        }
+        if( p_domain )
+        {
+            std::lock_guard< std::mutex > lock( p_domain->wait_mutex );
+            if( p_domain->seconds_to_wait > 0 )
+            {
+                // do this within the mutex: if multiple threads all try to query_devices, the others will
+                LOG_DEBUG( "waiting " << p_domain->seconds_to_wait << " seconds for devices on domain " << participant->domain_id() << " ..." );
+
+                // Set up a separate counter: if no new participants in the last 2 seconds, quit
+                auto listener = participant->create_listener();
+                std::atomic< int > seconds_left( 2 );
+                listener->on_participant_added( [&]( realdds::dds_guid, char const * ) { seconds_left = 2; } );
+
+                while( true )
+                {
+                    std::this_thread::sleep_for( std::chrono::seconds( 1 ) );
+                    if( --p_domain->seconds_to_wait <= 0 )
+                        break;
+                    if( --seconds_left <= 0 )
```

Comment:
I don't understand `seconds_left` purpose.
We need to wait `seconds_to_wait` seconds, which defaults to 5, why break after 2?

Reply:
Yes, maybe it's a little unclear -- and please verify my logic:
If we don't see any new participants, why keep waiting? The only way we'd see new devices is if we see new participants joining as we wait, right?
Participant discovery should be faster than for a device (which includes the handshake).
So if we wait 1.5 seconds and then see another participant joining, we'll extend our wait for another 2. If no new participants, we can return early.

Reply:
Going to split into two settings: `query-devices-min` and `query-devices-max`.
If max is 0, no wait will occur regardless of min.

Reply:
Default to 3-5?